frontend/console/src/features/modules/decls/DeclLink.tsx

```diff
@@ -0,0 +1,26 @@
+import { useMemo } from 'react'
+import { Link } from 'react-router-dom'
+import { useSchema } from '../../../api/schema/use-schema'
+import type { PullSchemaResponse } from '../../../protos/xyz/block/ftl/v1/ftl_pb.ts'
+
+export const DeclLink = ({ moduleName, declName }: { moduleName?: string; declName: string }) => {
+  const schema = useSchema()
+  const decl = useMemo(() => {
+    const modules = (schema?.data || []) as PullSchemaResponse[]
+    const module = modules.find((m: PullSchemaResponse) => m.moduleName === moduleName)
+    if (!module?.schema) {
+      return
+    }
+    return module.schema.decls.find((d) => d.value.value?.name === declName)
+  }, [moduleName, declName, schema?.data])
+
+  const str = moduleName ? `${moduleName}.${declName}` : declName
```

Comment:
I've been personally finding that it's easier to make "dumb" components and feed the data into them, vs. having to have `useSchema` type hooks in the smaller sub components. This also makes them much easier to write `storybook` stories for since they don't have any real dependencies. Whatcha think?

Reply:
Normally I follow that pattern too, but in this case, `DeclLink` is used by `TypeEl`, which in turn is used by most of the decl-type panels. We'd end up having to thread it through all the way, which would make the props for _all_ of those components more complicated just for _only_ `DeclLink` to actually use the contents. It gets worse from there, because `DeclLink` is probably going to end up all over the console once we're through, at which point we're basically passing `schema` down through nearly every single component in the tree. That would almost completely defeats the purpose of the hook pattern to begin with. :/

Reply:
Awesome! Yeah, definitely seems easier to keep this here then. Thanks for the clarification!